### PR TITLE
feat: add `next-version` subcommand

### PR DESCRIPTION
**Pull request by jstmlr**
This subcommand will examine the current latest tag in the repo,
and, if a semver-compatible version tag is found, returns a bumped
semver reflecting the changes (in the form of Conventional Commits)
_since_ that tag.

### CODE REVIEW

File: commisery/test/test_next_version.py [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+            "0.0.2",
+            0,
+        ),
+    ),
+)
+def test_next_version_generic(commisery_cli, commits, version, exitcode):
+    result = commisery_cli(commit_messages=commits, command="next-version", rev_range="")
+
+    print(f"Checking commits: {commits}")
+    # Test only up until the exlamation mark, if any
```
Comment (jstmlr) [inline commisery/test/test_next_version.py:133]: Left over from previous implementation; will remove.

File: commisery/test/test_next_version.py [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+                CommitAndTag("feat: test", None),
+            ),
+            "",
+            1,
+            ("HEAD^^",),
+        ),
+    ),
+)
+def test_next_version_target(commisery_cli, commits, version, exitcode, target):
+    print(f"Checking commits: {list(c.message for c in commits)}")
```
Comment (jstmlr) [inline commisery/test/test_next_version.py:164]: Will remove.

**Review by KevinDeJong-TomTom — COMMENTED** [review]
Please update the documentation

**Review by KevinDeJong-TomTom — APPROVED** [review]

**Comment by jstmlr** [comment]
Squashed away fixups.

**Review by KevinDeJong-TomTom — APPROVED** [review]

**Review by maikelvdh — DISMISSED** [review]

File: commisery/cli/__init__.py [hunk 3]
```diff
@@ -261,5 +252,115 @@ def check(ctx, target):
...
+        if repo_verify.active_branch.is_valid():
+            valid_repo = True
+    except InvalidGitRepositoryError:
+        pass
+
+    if not valid_repo:
+        log.error(f"Current directory ({os.getcwd()}) is not in a (valid) Git repository")
+        ctx.exit(1)
+
+    with Repo(search_parent_directories=True) as repo:
```
Comment (MartijnLeijssen-TomTom) [inline commisery/cli/__init__.py:296]: I guess you can reuse `repo_verify` here

**Review by MartijnLeijssen-TomTom — APPROVED** [review]

**Review by MartijnLeijssen-TomTom — DISMISSED** [review]

**Pull request merged**